src/mat.rs

```diff
@@ -1057,3 +1057,10 @@ macro_rules! mat4s {
 }
 
 mat4s!(Mat4 => Rotor3, Bivec3, Vec4, Vec3, f32, Wat4 => WRotor3, WBivec3, Wec4, Wec3, f32x4);
+
+// Utility functions for mat4 specific code
+impl Mat4 {
+    pub fn translate(&mut self, translation: &Vec3) {
```

Comment:
This is applicable to Wat4 too, no?

Reply:
I am getting immutability errors applying this via Wec/Wat for some reason, and I dont understand why.